src/Build/Evaluation/Evaluator.cs

```diff
@@ -2316,6 +2350,12 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
                         {
                             imports.Add(importedProjectElement);
 
+                            // Store a pointer to the newest project to be set later as MSBuildAllProjects
+                            if (importedProjectElement.TimeLastChanged > _newestProject.TimeLastChanged)
```

Comment:
I'm not sure why but this doesn't work as expected.  TimeLastChanged is not the last write time of the project?